contracts/modules/cover/Quotation.sol

```diff
@@ -275,78 +245,56 @@ contract Quotation is Iupgradable {
     return (a == qd.getAuthQuoteEngine());
   }
 
-  /**
-   * @dev to get the verdict of kyc process
-   * @param status is the kyc status
-   * @param _add is the address of member
-   */
-  function kycVerdict(address _add, bool status) public checkPause noReentrancy {
-    require(msg.sender == qd.kycAuthAddress());
-    _kycTrigger(status, _add);
-  }
-
-  /**
-   * @dev transfering Ethers to newly created quotation contract.
-   */
-  function transferAssetsToNewContract(address newAdd) public onlyInternal noReentrancy {
-    uint amount = address(this).balance;
-    IERC20 erc20;
-    if (amount > 0) {
-      // newAdd.transfer(amount);
-      Quotation newQT = Quotation(newAdd);
-      newQT.sendEther.value(amount)();
-    }
-    uint currAssetLen = pd.getAllCurrenciesLen();
-    for (uint64 i = 1; i < currAssetLen; i++) {
-      bytes4 currName = pd.getCurrenciesByIndex(i);
-      address currAddr = pd.getCurrencyAssetAddress(currName);
-      erc20 = IERC20(currAddr); // solhint-disable-line
-      if (erc20.balanceOf(address(this)) > 0) {
-        require(erc20.transfer(newAdd, erc20.balanceOf(address(this))));
-      }
-    }
-  }
-
-
   /**
    * @dev Creates cover of the quotation, changes the status of the quotation ,
    * updates the total sum assured and locks the tokens of the cover against a quote.
    * @param from Quote member Ethereum address.
    */
-
   function _makeCover(//solhint-disable-line
     address payable from,
-    address scAddress,
-    bytes4 coverCurr,
+    address contractAddress,
+    bytes4 coverCurrency,
     uint[] memory coverDetails,
     uint16 coverPeriod
-  )
-  internal
-  {
+  ) internal {
+
+    address underlyingToken = incidents.underlyingToken(contractAddress);
```

Comment:
since you called it productId there, and now it's an id not simply a smart contract address, it makes sense to do that here too